lib/Cake/Test/Case/Controller/Component/CookieComponentTest.php

```diff
@@ -790,9 +752,99 @@ public function testDeleteChildrenNotExist() {
 	}
 
 /**
+ * Test that deleting a child entry of the cookie array works
+ *
+ * @return void
+ */
+	public function testDeleteChildEntry() {
+		$response = $this->Controller->response;
+
+		$this->Cookie->write('User.name', 'user', false);
+		$this->Cookie->write('User.email', 'user@example.com', false);
```

Comment:
Good point. Also, from the viewpoint of security, it would be better not to expose any keys of the cookie if it is encrypted. There may be someone who want to store sensitive information to the key. As another approach, we can store expiration information to the cookie.